### ISSUE TITLE
possibility of adding groups GUID in acl_add_user()

### ISSUE THREAD

**Issue by mtcarsalot** [rstudio/connectapi#56]
Not sure if this is already possibly, but I cannot add groups to the acl access via the package. 

I can add individual users via the `connectapi:::acl_add_user()` function to a project, however I cannot add groups. 

This feature would help alot. 

great package!

**Comment by colearendt** [comment]
Thanks for reporting! You're right - I think this will need an `acl_add_group()` function or something like that.

I see you've found my unexported hackery 😉Has it been working well for your use? We are always happy to learn about the types of things that this package makes possible, especially since many of its functions represent unexported features of Connect 😅

**Comment by mtcarsalot** [comment]
I don't alway use :::, but when I do, I use it in production.

tl;dr: I need something like:
`acl_add_group(content, guid = NULL, username ="AD group username (not guid)")`

In connect, groups and users are added in the same field and to me they are kind of the same thing, but you do you. To make connect and apps compliant in our company (a bank) I have a source controled deployment script that is then deployed and processed by an AD service user that is an admin on the connect server, I don't have admin rights, not in prod atleast. Which mean I need to be able to configure everything via the deployment script and not through the connect interface. Here your vanity url addition is really nice, thanks!

Also; we have setup one AD group per application and we assign access by adding people to those groups. This works fine, but the IT department is tired of having to add the groups manually on new apps. So I would really like to have the access group in my deployment script aswell. (Also for version control of access). But for this to really work, there is a small bonus challenge: I must be able to add an AD group by its user name and not GUID, since there is a good change that, that group has not been used on the server before.

Great work on this package.

**Comment by colearendt** [comment]
Thanks for the background and detail here! It is a bit less than your ideal, but I think the current state of these APIs and the level of this package means that I shouldn't script too many layers of abstraction (yet). It should be pretty easy to write a little helper though:

```
con <- connect()

# create content
bnd <- bundle_dir("/my/deploy/app/path")
mycont <- deploy(con, bnd)

# create group
grp <- groups_create_remote(con, "support_group")
grp$guid[[1]]

# add acl
connectapi:::acl_add_group(mycont, grp$guid[[1]], "owner")

# get acl
get_acl_group(mycont)

# example wrapper
simple_wrapper <- function(connect, content_guid, group_name, role = "viewer") {
  grp <- groups_create_remote(connect, group_name)
  acl_add_group(content_item(connect, content_guid), grp$guid[[1]], role = role)
}

# using the wrapper
simple_wrapper(con, mycont$get_content()$guid, "support_group", "viewer")
simple_wrapper(con, mycont$get_content()$guid, "finance_group", "owner")
```

Please let me know if you run into any issues, or if you have any additional feedback on Connect / the API and the types of use cases it unlocks for you! 😄
